DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/CompetencyDataService.cs

```diff
@@ -0,0 +1,532 @@
+﻿namespace DigitalLearningSolutions.Data.DataServices.SelfAssessmentDataService
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Models.Frameworks;
+    using DigitalLearningSolutions.Data.Models.SelfAssessments;
+    using Microsoft.Extensions.Logging;
+    using AssessmentQuestion = DigitalLearningSolutions.Data.Models.SelfAssessments.AssessmentQuestion;
+
+    public partial class SelfAssessmentDataService
+    {
+        private const string LatestAssessmentResults =
+            @"LatestAssessmentResults AS
+            (
+                SELECT
+                    s.CompetencyID,
+                    s.AssessmentQuestionID,
+                    s.ID AS ResultID,
+                    s.Result,
+                    s.SupportingComments,
+                    sv.ID AS SelfAssessmentResultSupervisorVerificationId,
+                    sv.Requested,
+                    sv.Verified,
+                    sv.Comments,
+                    sv.SignedOff,
+                    0 AS UserIsVerifier,
+                    COALESCE (rr.LevelRAG, 0) AS ResultRAG
+                FROM SelfAssessmentResults s
+                INNER JOIN (
+                    SELECT MAX(ID) as ID
+                    FROM SelfAssessmentResults
+                    WHERE CandidateID = @candidateId
+                        AND SelfAssessmentID = @selfAssessmentId
+                    GROUP BY CompetencyID, AssessmentQuestionID
+                ) t
+                    ON s.ID = t.ID
+                LEFT OUTER JOIN SelfAssessmentResultSupervisorVerifications AS sv
+                    ON s.ID = sv.SelfAssessmentResultId AND sv.Superceded = 0
+                LEFT OUTER JOIN CompetencyAssessmentQuestionRoleRequirements rr

```

Comment:
Is it possible to break longer lines like this one onto multiple lines?

Reply:
Broke down some of the longer lines in this file.